### PR TITLE
Cross-build(scala 2.11) for ergoWallet; Automate publishing; Remove sbt-lock;

### PR DESCRIPTION
**Pull request by greenhat**
Removed sbt-lock due to incompatibility with cross-builds [link](https://github.com/tkawachi/sbt-lock/issues/11)

Wallet publishing setup like in https://github.com/ScorexFoundation/sigmastate-interpreter/pull/619

Version is set via git tag vX.Y.Z:
```  
$ git tag v3.2.0
$ git push origin v3.2.0
```
Without the tag version resolves to `[branch name]-[git commit hash]-SNAPSHOT`.

When git tag is set Travis will run a release stage publishing `ergoWallet`. If not git tag is set Travis will publish a snapshot version of `ergoWallet`.

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -33,16 +33,16 @@ libraryDependencies ++= Seq(
   ("org.scorexfoundation" %% "scorex-core" % scorexVersion).exclude("ch.qos.logback", "logback-classic"),
 
   "org.typelevel" %% "cats-free" % "1.6.0",
-  "javax.xml.bind" % "jaxb-api" % "2.+",
-  "com.iheart" %% "ficus" % "1.4.+",
+  "javax.xml.bind" % "jaxb-api" % "2.4.0-b180830.0359",
+  "com.iheart" %% "ficus" % "1.4.7",
```
Comment (aslesarenko) [inline build.sbt]: Why this is needed?

File: ergo-wallet/build.sbt [hunk 2]
```diff
@@ -1,3 +1,15 @@
+/*
+### How to publish snapshot version fo ergoWallet:
```
Comment (aslesarenko) [inline ergo-wallet/build.sbt]:
```suggestion
### How to publish snapshot version of ergoWallet:
```

File: ergo-wallet/src/test/scala/org/ergoplatform/wallet/interpreter/ErgoUnsafeProverSpec.scala [hunk 3]
```diff
@@ -94,7 +94,8 @@ class ErgoUnsafeProverSpec
     override def sigmaLastHeaders: Coll[Header] = Colls.emptyColl
 
     override def previousStateDigest: ADDigest = Base16.decode("a5df145d41ab15a01e0cd3ffbab046f0d029e5412293072ad0f5827428589b9302")
-      .fold(_ => throw new Error(s"Failed to parse genesisStateDigest"), ADDigest @@ _)
+      .map(ADDigest @@ _)
+      .getOrElse(throw new Error(s"Failed to parse genesisStateDigest"))
```
Comment (aslesarenko) [inline ergo-wallet/src/test/scala/org/ergoplatform/wallet/interpreter/ErgoUnsafeProverSpec.scala]: Yeah, this is much easier than what I did in sigma.

File: build.sbt (hunk 1 above)
Comment (greenhat) [reply build.sbt]: @aslesarenko It's not needed, I just "moved" here the exact versions that were in `lock.sbt`.

**Pull request merged**